common/utils/src/main/java/org/graalvm/buildtools/utils/NativeImageUtils.java

```diff
@@ -122,7 +122,10 @@ public static void checkVersion(String requiredVersion, String versionToCheck) {
         if (versionToCheck.contains("GraalVM Runtime Environment")) {
             return; // later than 22.3.1 (e.g., GraalVM for JDK 17 / GraalVM for JDK 20)
         }
-        if (versionToCheck.startsWith("GraalVM dev") || versionToCheck.startsWith("native-image dev")) {
+        if (!versionToCheck.startsWith("GraalVM") && !versionToCheck.startsWith("native-image") && !versionToCheck.startsWith("Oracle GraalVM")) {
+            throw new IllegalArgumentException("Invalid version " + versionToCheck + ", should start with GraalVM, Oracle GraalVM or native-image");
+        }
+        if (versionToCheck.contains("dev")) {
```

Comment:
Oh, I think we can simplify this as follows:

```suggestion
        if ((versionToCheck.startsWith("native-image") && versionToCheck.contains("-dev+")) ||
                 versionToCheck.startsWith("GraalVM dev")) { /* For GraalVM 22.3 and earlier */
            return;
        }
```

Reply:
This is a bit stricter than it was in last commit, but as long as we don't introduce some new format, this will work.

Reply:
I didn't like `versionToCheck.contains("dev")` in particular. What if, for some reason, a word like "developer" becomes part of the version string?